### ISSUE TITLE
Feature request: Backend function akin to tf.split

### ISSUE THREAD

**Issue by flomlo** [fchollet/keras#7718]
Hello!

I've been using tf.split [1] now several times in my own keras libraries. This sucks, as I want to programm backend-neutral libraries. Did I miss an obvious way to do that in Keras? Please let me know then.

I also consider the functionality of tf.split not an edge case, especially the ability to input a tensor and produce different-sized-slices is very useful.

I therefore propose adding keras.backend.split with functionality as given by [1].

Would that be considered bloating the API? 

If not, I could probably figure out how to do that for both the tensorflow backend (trivial) and the Theano backend (might require a bit of work).
Unfortunately I know nothing about the CNTK-backend, so help would be appreciated there.

[1] https://www.tensorflow.org/api_docs/python/tf/split

==============================

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

**Comment by kgrm** [comment]
Equivalent functionality is already trivially achievable in a backend-agnostic fashion using tensor indexing.

```
>>> import keras.backend as K
Using Theano backend.
>>> x = K.variable(np.random.rand(4, 15, 11))
>>> t1, t2, t3 = [x[:, 5*i : 5*i + 5, :] for i in xrange(3)]
>>> t1.eval().shape
(4, 5, 11)
>>> t2.eval().shape
(4, 5, 11)
>>> t3.eval().shape
(4, 5, 11) 
```

**Comment by flomlo** [comment]
Thanks you for your answer!

I agree that we can easily do stuff like that ourself for easy slicing operations, like getting equally sized chunks.
But what about splitting some tensor in subtensors of size of e.g. (20, 1, 1, 1, 3, 5, 9, 3)? In that case it gets brutal very quickly.

**Comment by flomlo** [comment]
Damn, I'm an idiot, now I see.

Sorry for (unintentionally) abusing the bugtracker as stackoverflow!
I will mark it as resolved.

I will implement and test the code I have in mind, then post it here in case anyone else has the question.

**Comment by agamemnonc** [comment]
Any updates on this?

**Comment by dqii** [comment]
^

**Comment by maitreya2954** [comment]
@kgrm slicing the tensor is reducing the dimension of the tensor. Any idea on what am I doing wrong?

            [c_IJ = self.softmax(b_IJ, axis=2)
            print('softmax: ', K.int_shape(c_IJ))
            c_Ij = c_IJ[:,:,idx_j,:]
            print(c_Ij)](url)
Output:
softmax:  (None, 1152, 5, 1)
Tensor("capsule2d_1/strided_slice_3:0", shape=(?, 1152, 1), dtype=float32)
